### PR TITLE
SPIR-V Environment Spec Refactor

### PR DESCRIPTION
**Pull request by bashbaug**
This change refactors the SPIR-V environment spec to increase readability, reduce duplicated information, and break the tight coupling between an OpenCL version and the SPIR-V version(s) it supports.

I've fixed a few bugs as well, see #80 and #81.

### CODE REVIEW

File: env/common_properties.asciidoc [hunk 1]
```diff
@@ -196,18 +154,18 @@ The following table describes other data types that may be used in an
...
 
 // event_t
 | *OpTypeEvent*
-| OpenCL event type.
+| OpenCL event representing async copies from global to local memory and vice-versa.
 
 // clk_event_t
 | *OpTypeDeviceEvent*
-| OpenCL device-side event type.
+| OpenCL device-side event representing commands enqueued to a device command queues.
```
Comment (kpet) [inline env/common_properties.asciidoc]: typo

File: env/introduction.asciidoc [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+<<opencl-spec, OpenCL>> is an open, royalty-free, standard for general
+purpose parallel programming across CPUs, GPUs, and other processors, giving
+software developers portable and efficient access to the power of
+heterogeneous processing platforms.
+
+<<spirv-spec, SPIR-V>> is an open, royalty-free, standard intermediate
+language capable of representing parallel compute kernels that are executed
+by implementations of the OpenCL standard.
+
+SPIR-V is adaptable to multiple execution environments: A SPIR-V module is
```
Comment (kpet) [inline env/introduction.asciidoc]: s/A/a/

File: env/introduction.asciidoc [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+This document describes the SPIR-V execution environment for implementations
+of the OpenCL standard.
+The SPIR-V execution environment describes required support for some SPIR-V
+capabilities, additional semantics for some SPIR-V instructions, and
+additional validation rules that a SPIR-V binary module must adhere to in
+order to be considered valid.
+
+This document is written for compiler developers who are generating SPIR-V
+modules intended to be consumed by the OpenCL API, for implementors of the
+OpenCL API who are consuming SPIR-V modules, and by software developers who
```
Comment (kpet) [inline env/introduction.asciidoc]: Did you intend to write "by"?

File: env/common_properties.asciidoc (hunk 1 above)
Comment (bashbaug) [reply env/common_properties.asciidoc]: Fixed, thanks!

File: env/introduction.asciidoc (hunk 2 above)
Comment (bashbaug) [reply env/introduction.asciidoc]: Fixed, thanks!

File: env/introduction.asciidoc (hunk 3 above)
Comment (bashbaug) [reply env/introduction.asciidoc]: Fixed: by -> for.

**Comment by bashbaug** [comment]
Thanks @kpet, I believe I've addressed all of your comments.  Note, I've opened #88 regarding one part of the 64-bit atomics section, but I don't think that should hold up these changes.

**Comment by kpet** [comment]
Thanks! Well-spotted.Yes, it shouldn't hold up this PR (commenting on the Int64Atomics capability was already a bit of a stretch).

**Pull request merged**